lib/beaker/puppet_install_helper.rb

```diff
@@ -31,7 +31,8 @@ def run_puppet_install_helper_on(hosts, type_arg = find_install_type, version =
     case type
     when 'pe'
       # These will skip hosts that are not supported
-      install_pe_on(Array(hosts), options.merge('pe_ver' => version))
+      install_pe_on(Array(hosts), options.merge('pe_ver' => ENV['BEAKER_PE_VER'] || ENV['PUPPET_INSTALL_VERSION'],
```

Comment:
Setting `pe_ver` to a `PUPPET_INSTALL_VERSION` like `5.4.0` would fail provisioning wouldn't it?

Reply:
Yep, and that's actually what's currently happening with `BEAKER_PUPPET_AGENT_VERSION`:
```
06:51:23 + export BEAKER_PE_VER=2017.3.6-rc0-90-g7021def
06:51:23 + export PUPPET_INSTALL_VERSION=2017.3
06:51:23 + export BEAKER_PUPPET_AGENT_VERSION=5.3.5
[...]
06:52:15   Don't know how to generate answers for 5.3.5
```

I didn't write BPIH to take a `PUPPET_INSTALL_VERSION` that was a different number than the agent (back then PE and agents were versions like 3.4.0). When the 2016.x PE versions diverged from AIO 4.x versions someone figured out that they can pass PE versions to `PUPPET_INSTALL_VERSION` and agent versions to `BEAKER_PUPPET_AGENT_VERSION` to get around that instead of changing BPIH to accept both of them.

`PUPPET_INSTALL_TYPE` and `PUPPET_INSTALL_VERSION` are deprecated in favor of `BEAKER_IS_PE`, `BEAKER_PUPPET_COLLECTION`, and `BEAKER_PUPPET_AGENT_VERSION` (see 0.9.0 changelog). But I wasn't aware of the above hacky workaround in ci-job-configs so am now trying to make a workable solution.

I didn't have `PUPPET_INSTALL_VERSION` here originally and could take it out again. I think CI would be okay with that. I only added it back in case *someone* is still using it, but if it's deprecated then it shouldn't be catered to I guess?

Reply:
Personally, I'd prefer it be removed. If someone is using `PUPPET_INSTALL_VERSION` incorrectly as `pe_ver`, it would probably(?) be easier to debug the provision error that says `pe_ver` is nil, than it would be to chase their tails wondering what ENV variable is missing.